### PR TITLE
build/tools/amebasmart: Memory Layout Changes & Standardize compiler flag for Optimization

### PR DESCRIPTION
**Pull request by ZhenBei-Sin**
build/tools/amebasmart: Standardize compiler flag
Changes Note:
- Change optimizer level from "-O2" to "-Os"
- Change RTK pre-build lib toolchain flag from "-marm" to "-mthumb"
- Removed unused lib ble_mgr.a & bt_mgr.a & leaudio.a

build/configs/rtl8730e: Adjust Memory Layout
Changes Note:
Memory Optimization
- Remove CA32 SRAM_NOCACHE_DATA_SECTION as sd/emmc is not used
- Increase reserved heap around 3KB

Adjust PSRAM layout
- CA32 ATF address adjusted: 0x7001_7000 - 0x7010_0000; 932KB
- CA32 BL33 address adjusted: 0x6010_0000 - 0x6080_0000; 7M

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@jylee9613 could you check some tests with this?

**Comment by jylee9613** [comment]
@ZhenBei-Sin I tested this PR in our product repo, but hang occurs in non-signing board. (There's no hang in signing board)
The location of hang is random, but hang occurs after connect with micom.

**Comment by ZhenBei-Sin** [comment]
> @ZhenBei-Sin I tested this PR in our product repo, but hang occurs in non-signing board. (There's no hang in signing board) The location of hang is random, but hang occurs after connect with micom.

Hi @jylee9613 , will be check with Eric for further test on it.
Please attach the hang log. 
Thank You.

File: build/tools/amebasmart/bootloader/km4_boot_RELEASE_NOTE.txt [hunk 1]
```diff
@@ -1,5 +1,11 @@
 /* == "version" + "Realtek git version" + "compile date" + "compile time" == */
 
+== version 2d0f1be032 2024/04/29-09:51:17 ==
+1.	Change optimization flags from "-O2" to "-Os"
+2.	Adjust the PSRAM Layout
+	- CA32 ATF address adjusted: 0x7018_0000 - 0x7010_0000; 932KB
```
Comment (edwakuwaku) [inline build/tools/amebasmart/bootloader/km4_boot_RELEASE_NOTE.txt:6]: ATF: 0x7001_7000~0x7010_0000

**Review by sunghan-chang — COMMENTED** [review]
`Memory Layout change` and `Standardize compiler flag for Optimization` are different topics. Please split the commit as two.

**Comment by ZhenBei-Sin** [comment]
> `Memory Layout change` and `Standardize compiler flag for Optimization` are different topics. Please split the commit as two.

Hi @sunghan-chang ,

The memory layout changes and standardize compiler flag was difficult to split it.
As the both changes was affected the pre-build img.

**Comment by sunghan-chang** [comment]
@ZhenBei-Sin Ok, then please add more explanation of the changes about memory layout change.
```
Adjust PSRAM layout

CA32 ATF address adjusted: 0x7001_7000 - 0x7010_0000; 932KB
CA32 BL33 address adjusted: 0x6010_0000 - 0x6080_0000; 7M
```

**Comment by ZhenBei-Sin** [comment]
> @ZhenBei-Sin Ok, then please add more explanation of the changes about memory layout change.
> 
> ```
> Adjust PSRAM layout
> 
> CA32 ATF address adjusted: 0x7001_7000 - 0x7010_0000; 932KB
> CA32 BL33 address adjusted: 0x6010_0000 - 0x6080_0000; 7M
> ```

Hi @sunghan-chang 
We will explain more details in the email.

**Comment by jylee9613** [comment]
Hello @ZhenBei-Sin. Sorry for late reply. I attached hang log throgh the email. Thank you.

**Comment by sunghan-chang** [comment]
@ZhenBei-Sin Could you resolve the conflicts ASAP? When it's done, let me merge this with PR #6102 .

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**